File: Core.DomainModel/ItProject/GoalStatus.cs
using System;
using System.Collections.Generic;

namespace Core.DomainModel.ItProject
{
    /// <summary>
    /// It project goal status.
    /// </summary>
    public class GoalStatus : Entity, IProjectModule, ISupportsUserSpecificAccessControl
    {
        public GoalStatus()
        {
            this.Goals = new List<Goal>();
        }

        public virtual ItProject ItProject { get; set; }

        /// <summary>
        /// Traffic-light dropdown for overall status.
        /// </summary>
        /// <value>
        /// The status.
        /// </value>
        public TrafficLight Status { get; set; }
        /// <summary>
        /// Date-for-status-update field
        /// </summary>
        public DateTime? StatusDate { get; set; }

        /// <summary>
        /// Notes on collected status on project
        /// </summary>
        public string StatusNote { get; set; }

        public virtual ICollection<Goal> Goals { get; set; }

        public bool HasUserWriteAccess(User user)
        {
            return ItProject != null && ItProject.HasUserWriteAccess(user);
        }
    }
}

File: Presentation.Web/app/models/it-project/goal-status.ts
module Kitos.Models.ItProject {
	/** It project goal status. */
    export interface IGoalStatus extends IEntity {
        ItProject: IItProject;
        /** Traffic-light dropdown for overall status. */
        Status: TrafficLight;
        /** Date-for-status-update field */
        StatusDate: Date;
        /** Notes on collected status on project */
        StatusNote: string;
        Goals: Array<IGoal>;
    }
}
